### PR TITLE
Jsonschema update

### PR DESCRIPTION
**Pull request by willronchetti**
- Updates fourfront to use new `jsonschema` standard
- Improves some validation messages in specific scenarios
- Repairs user registration

### CODE REVIEW

**Review by dmichaels-harvard — APPROVED** [review]

File: src/encoded/tests/test_fixtures.py [hunk 1]
```diff
@@ -98,6 +98,7 @@ def test_fixtures2(minitestdata2, testapp):
     assert len(items) == 1
 
 
+@pytest.mark.skip  # not clear this has been working for some time
```
Comment (netsettler) [inline src/encoded/tests/test_fixtures.py:101]: Not sure I get this. Wouldn't it have been erring?

File: src/encoded/tests/test_permissions.py [hunk 2]
```diff
@@ -368,8 +368,9 @@ def test_submitter_post_update_experiment(submitter_testapp, lab, award, human_b
 def test_submitter_cant_post_other_lab(submitter_testapp, other_lab, award, exp_types):
     experiment = {'lab': other_lab['@id'], 'award': award['@id'], 'experiment_type': exp_types['microc']['@id']}
     res = submitter_testapp.post_json('/experiments-hi-c', experiment, status=422)
-    assert res.json['errors'][0]['name'] == 'Schema: lab'
-    assert "not in user submits_for" in res.json['errors'][0]['description']
+    # jsonschema - error ordering change
+    assert res.json['errors'][1]['name'] == 'Schema: lab'
+    assert "not in user submits_for" in res.json['errors'][1]['description']
```
Comment (netsettler) [inline src/encoded/tests/test_permissions.py:373]: Maybe something like this (untested):
```suggestion
    assert any(error['name'] == 'Schema: lab' and "not in user submits_for" in error['description']
               for error in res.json['errors'])
```

File: src/encoded/tests/test_post_put_patch.py [hunk 3]
```diff
@@ -208,9 +208,10 @@ def test_post_object_with_child(content_with_child, testapp):
         ]
     }
     res = testapp.post_json('/testing-link-targets', edit, status=422).json
-    assert len(res['errors']) == 1
-    assert res['errors'][0]['description'] == 'submission of calculatedProperty disallowed'
-    assert res['errors'][0]['name'] == 'Schema: reverse'
+    errors = sorted(res['errors'], key=lambda d: d['description'])
+    assert len(errors) == 2
+    assert errors[1]['description'] == 'submission of calculatedProperty disallowed'
+    assert errors[1]['name'] == 'Schema: reverse'
```
Comment (netsettler) [inline src/encoded/tests/test_post_put_patch.py:214]: Maybe something like this (untested):
```suggestion
    assert any(error['name'] == 'Schema: reverse' and error['description'] == 'submission of calculatedProperty disallowed'
               for error in res.json['errors'])
```

File: src/encoded/tests/testing_views.py [hunk 4]
```diff
@@ -153,7 +153,7 @@ def rev_link_atids(self, request, rev_name):
         "type": "array",
         "items": {
             "type": ['string', 'object'],
-            "linkTo": "TestingLinkSourceSno",
+            "linkTo": "TestingLinkSource",
```
Comment (netsettler) [inline src/encoded/tests/testing_views.py:156]: What was this change about?

**Review by netsettler — APPROVED** [review]
My comments are not blocking.

File: src/encoded/tests/testing_views.py (hunk 4 above)
Comment (willronchetti) [reply src/encoded/tests/testing_views.py:156]: This is a bug in the definition that was not throwing a validation error previously but does now with the better validation

**Review by aschroed — COMMENTED** [review]
Validation is failing on this Ontology master insert:
https://github.com/4dn-dcic/fourfront/blob/a906c19472609f5510108e114867c655b1590a5e/src/encoded/tests/data/master-inserts/ontology.json#L28
needs the https: prefix.

**Review by aschroed — COMMENTED** [review]
A couple more non-valid URLs:
https://github.com/4dn-dcic/fourfront/blob/a906c19472609f5510108e114867c655b1590a5e/src/encoded/tests/data/inserts/quality_metric_rnaseq_madqc.json#L32

https://github.com/4dn-dcic/fourfront/blob/a906c19472609f5510108e114867c655b1590a5e/src/encoded/tests/data/inserts/quality_metric_workflowrun.json#L6

**Review by aschroed — APPROVED** [review]

**Pull request merged**